lib/synapse/haproxy.rb

```diff
@@ -717,7 +717,7 @@ def generate_backend_stanza(watcher, config)
         config.map {|c| "\t#{c}"},
         backends.keys.shuffle.map {|backend_name|
           backend = backends[backend_name]
-          b = "\tserver #{backend_name} #{backend['host']}:#{backend['port']}"
+          b = "\tserver #{backend_name} #{backend['host']}:#{backend['port']}#{' backup' if backend['backup']}#{' check port ' if backend['httpchk']}#{backend['chkport']}"
```

Comment:
Thoughts on having a ``haproxy_server_options`` catch all registration info for backend registrations similar to the watcher's ``server_options``?

So like, instead of ``haproxy_backup`` and ``haproxy_httpchk`` you just have ``haproxy_server_options`` which can contain weights, httpcheck lines, ports, backup status, etc ...